### PR TITLE
[mlir][ArmSME] Rename slice move operations to insert/extract_tile_slice

### PR DESCRIPTION
**Pull request by MacDue**
This renames:

- `arm_sme.move_tile_slice_to_vector` to `arm_sme.extract_tile_slice`
- `arm_sme.move_vector_to_tile_slice` to `arm_sme.insert_tile_slice`

The new names are more consistent with the rest of MLIR and should be easier to understand. The current names (to me personally) are hard to parse and easy to mix up when skimming through code.

Additionally, the syntax for `insert_tile_slice` has changed from:

```mlir
%4 = arm_sme.insert_tile_slice %0, %1, %2
  : vector<[16]xi8> into vector<[16]x[16]xi8>
```

To:

```mlir
%4 = arm_sme.insert_tile_slice %0, %1[%2]
  : vector<[16]xi8> into vector<[16]x[16]xi8>
```

This is for consistency with `extract_tile_slice`, but also helps with readability as it makes it clear which operand is the index.

### CODE REVIEW

**Review by banach-space — APPROVED** [review]
LGTM, thanks Ben!

File: mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td [hunk 1]
```diff
@@ -603,25 +603,25 @@ def MoveVectorToTileSliceOp : ArmSME_Op<"move_vector_to_tile_slice", [
...
-  let summary = "Move 1-D scalable vector to slice of 2-D tile";
+  let summary = "Insert 1-D scalable vector into slice of 2-D tile";
   let description = [{
-    The vector to tile slice operation moves a 1-D scalable vector to a slice
-    of a 2-D scalable vector tile at the given index. The type of the 1-D
-    scalable vector to be moved must match the type of the tile slice. A tile
-    slice is a 1-D vector of horizontally or vertically contiguous elements
-    within a ZA tile. The updated tile is returned as the result.
+    Inserts a 1-D scalable vector to a slice of a 2-D scalable vector tile at
+    the given index. The type of the 1-D scalable vector to be moved must match
```
Comment (c-rhodes) [inline mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td:609]:
```suggestion
    the given index. The type of the 1-D scalable vector to be inserted must match
```

File: mlir/lib/Conversion/ArmSMEToSCF/ArmSMEToSCF.cpp [hunk 2]
```diff
@@ -245,7 +245,7 @@ struct TileLoadOpConversion : public OpRewritePattern<arm_sme::TileLoadOp> {
 ///      : memref<?x?xi32>, vector<[4]xi1>,
 ///        vector<[4]xi32> into vector<[4]xi32>
 ///    // Insert slice into tile
-///    %tile_update = arm_sme.move_vector_to_tile_slice
+///    %tile_update = arm_sme.insert_tile_slice
 ///      %slice, %iter_tile, %tile_slice_idx :
```
Comment (c-rhodes) [inline mlir/lib/Conversion/ArmSMEToSCF/ArmSMEToSCF.cpp:249]: syntax needs updating

File: mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp [hunk 3]
```diff
@@ -199,7 +199,7 @@ struct VectorStoreToArmSMELowering : public OpRewritePattern<vector::StoreOp> {
 ///   %broadcast_to_tile = scf.for %tile_slice_index = %c0 to %num_tile_slices
 ///       step %c1 iter_args(%iter_tile = %init_tile) -> (vector<[4]x[4]xi32>)
 ///   {
-///     %tile_update = arm_sme.move_vector_to_tile_slice
+///     %tile_update = arm_sme.insert_tile_slice
 ///        %broadcast_to_1d, %iter_tile, %tile_slice_index :
```
Comment (c-rhodes) [inline mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp:203]: syntax needs updating

File: mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp [hunk 4]
```diff
@@ -267,7 +267,7 @@ struct BroadcastOpToArmSMELowering
...
```
Comment (c-rhodes) [inline mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp:271]: syntax needs updating

**Review by c-rhodes — APPROVED** [review]
Few minor comments but otherwise LGTM cheers
